Assets/Plugins/Editor/Uplift/Schemas/Repositories/GithubRepository.cs

```diff
@@ -0,0 +1,128 @@
+    {
+        private GitHubRelease release;
+
+        public override TemporaryDirectory DownloadPackage(Upset package)
+        {
+            string sourceName = Regex.Replace(package.MetaInformation.dirName, ".Upset.xml$", ".unitypackage", RegexOptions.IgnoreCase);
+
+            release = GetPackagesRelease();
+            if(!release.assets.Any(asset => asset.name.Contains(sourceName)))
+                throw new ArgumentException(string.Format("Package {0} is not present in this repository", package.PackageName));
+
+            GitHubAsset packageAsset = release.assets.First(asset => asset.name.Contains(sourceName));
+            TemporaryDirectory td = new TemporaryDirectory();
+            StreamReader sr = new StreamReader(GitHub.GetAssetStream(packageAsset, GetToken()));
+            UnityPackage unityPackage = new UnityPackage();
+            unityPackage.Extract(sr.BaseStream, td.Path);
+            return td;
+        }
+
+        public override Upset[] ListPackages()
+        {
+            release = GetPackagesRelease();
+
+            GitHubAsset[] upsetAssets = release.assets.Where(asset => asset.name.EndsWith("Upset.xml")).ToArray();
+
+            string progressBarTitle = "Parsing Upsets from GitHub repository";
+            int index = 0;
+
+            List<Upset> upsetList = new List<Upset>();
+            EditorUtility.DisplayProgressBar(progressBarTitle, "Please wait a little bit while Uplift parses the Upset in the GitHub repository at " + urlField, (100f * (float)index) / upsetAssets.Length);
+            foreach(GitHubAsset asset in upsetAssets)
+            {
+                StrictXmlDeserializer<Upset> deserializer = new StrictXmlDeserializer<Upset>();
+                
+                StreamReader sr = new StreamReader(GitHub.GetAssetStream(asset, GetToken()));
+                UnityPackage unityPackage = new UnityPackage();
+                Upset upset = deserializer.Deserialize(sr.BaseStream);
+                upset.MetaInformation.dirName = asset.name;
+                upsetList.Add(upset);
+                EditorUtility.DisplayProgressBar(
+                    progressBarTitle,
+                    "Parsing " + asset.name, 
+                    (100f * (float)(++index)) / upsetAssets.Length
+                );
+            }
+
+            EditorUtility.ClearProgressBar();
+            return upsetList.ToArray();
+        }
+
+        private GitHubRelease GetPackagesRelease()
+        {
+            if(release == null)
+            {
+                IEnumerator e = GitHub.LoadReleases(urlField, GetToken());
+                do { Thread.Sleep(1000); } while (e.MoveNext());
+
+                GitHubRelease[] releases = (GitHubRelease[]) e.Current;
+                if (releases == null)
+                    throw new ApplicationException("This github repository does not have releases");
+
+                if (!releases.Any(rel => rel.tag == "packages"))
```

Comment:
You could use `FirstOrDefault(..., null)` and fail if result if null (to not have to match for `Any` first)